source/logHandler.py

```diff
@@ -141,15 +141,21 @@ def getCodePath(f):
 	return ".".join(x for x in (path,className,funcName) if x)
 
 
+_onErrorSoundRequested = None
+
+
+def getOnErrorSoundRequested():
+	global _onErrorSoundRequested

```

Comment:
In my initial proposal I have suggested to use `functools.lru_cache` to cache the created extension point, which saves you from implementing the caching by hand, and ensures the getter function is indeed only possible way to access it. While its not something I feel strongly about, I'm just curious if there is a particular reason why you went with the module level global approach here?

Reply:
I have used the private global because it's done this way elsewhere in NVDA to retrieve module level variables (even if it's not for the same reasons); the 1-value cache instead does not seem to be used in NVDA's code base. Thus the whole code base seems more consistent this way.

Also, I have not investigated if the content of the cache is easily accessible for debug purpose if needed.

If NV Access want me to use the cache instead as a new recommended pattern, no problem, I'll change it.

Reply:
The suggested patterns makes sense, but the current pattern is fine also